Source/KSCrash/Recording/Tools/KSCPU.c

```diff
@@ -31,15 +31,27 @@
 
 #include <mach/mach.h>
 #include <mach-o/arch.h>
+#include <mach-o/utils.h>
 
 //#define KSLogger_LocalLevel TRACE
 #include "KSLogger.h"
 
 
 const char* kscpu_currentArch(void)
 {
-    const NXArchInfo* archInfo = NXGetLocalArchInfo();
-    return archInfo == NULL ? NULL : archInfo->name;
+#if !KSCRASH_HOST_VISION
+    if(__builtin_available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 8.0, *))
```

Comment:
visionOS seems to not being supported in `@available` and `__builtin_available`. So for it I just use new API without any checks.